src/base/bittorrent/session.cpp

```diff
@@ -1124,6 +1124,7 @@ void Session::configure(libtorrent::settings_pack &settingsPack)
             }
         }
 
+        settingsPack.set_str(libt::settings_pack::outgoing_interfaces, networkInterface().toStdString());
```

Comment:
Are you sure that @Chocobo1 isn't right? Look inside `Session::getListeningIPs()`. We do some sanity checks there. Some codepaths return only one specific IP address. I think in those cases the outgoing interface should be bound to the same thing as the listening interface is.